File: 프로그래머스/unrated/172928. 공원 산책/공원 산책.py
def solution(park, routes):
    answer = []
    s = []
    x = len(park[0])
    y = len(park)
    for i in range(y):
        for j in range(x):
            if park[i][j] == 'S':
                s = [i, j]
                break
        if len(s) == 2:
            break
    oper = {'N' : [-1, 0], 'S' : [1, 0], 'W' : [0, -1], 'E' : [0, 1]}
    for data in routes:
        op, n = data.split()
        n_y, n_x = oper[op]
        d = 0
        n = int(n)
        for j in range(1, n + 1):
            dy, dx = s[0] + n_y * j , s[1] + n_x * j
            if dy < 0 or dy >= y or dx < 0 or dx >= x:
                d = 1
                break
            elif park[dy][dx] == 'X':
                d = 1
                break
                
        if d == 0:
            s[0] = s[0] + n_y * n
            s[1] = s[1] + n_x * n
    
    
    return s

File: 프로그래머스/1/172928. 공원 산책/공원 산책.js
function solution(park, routes) {
    var answer = [];
    const x_len = park[0].length;
    const y_len = park.length;
    for (let i = 0; i < y_len; i++){
        for (let j = 0; j < x_len; j++){
            if (park[i][j] === 'S') {
                answer.push(i);
                answer.push(j);
                break;
            };
        };
    };
    const op = {'N': [-1, 0], 'S': [1, 0],'W': [0, -1],'E': [0, 1]}
    for (let i = 0; i < routes.length; i++){
        const o = op[routes[i][0]];
        const c = Number(routes[i][2]);
        let t = 0
        if (0 <= answer[0] + o[0] * c && answer[0] + o[0] * c < y_len && 0 <= answer[1] + o[1] * c && answer[1] + o[1] * c < x_len){
            for (let k = 1; k < c + 1; k++){
                if (park[answer[0] + o[0] * k][answer[1] + o[1] * k] === 'O' || park[answer[0] + o[0] * k][answer[1] + o[1] * k] === 'S'){
                    t++
                };
            };  
            if (t === c){
                answer[0] = answer[0] + o[0] * c
                answer[1] = answer[1] + o[1] * c
            };
        };
    };
    return answer;
}